examples/sycl/09_bmg_grouped_gemm_fp8/09_bmg_grouped_gemm_fp8.cpp

```diff
@@ -0,0 +1,673 @@
+  using ElementC = typename Gemm::ElementC;
+
+  using LayoutA = typename Gemm::LayoutA;
+  using LayoutB = typename Gemm::LayoutB;
+  using LayoutC = typename Gemm::LayoutC;
+  using LayoutD = typename Gemm::LayoutD;
+
+  using CollectiveEpilogue = typename Gemm::CollectiveEpilogue;
+  using ElementOutput = typename CollectiveEpilogue::ElementOutput;
+  using ElementAccumulator = ElementOutput;
+
+  using StrideA = typename Gemm::GemmKernel::InternalStrideA;
+  using StrideB = typename Gemm::GemmKernel::InternalStrideB;
+  using StrideC = typename Gemm::GemmKernel::InternalStrideC;
+  using StrideD = typename Gemm::GemmKernel::InternalStrideD;
+
+  // Host-side allocations
+  std::vector<int64_t> offset_A;
+  std::vector<int64_t> offset_B;
+  std::vector<int64_t> offset_C;
+  std::vector<int64_t> offset_D;
+
+  std::vector<StrideA> stride_A_host;
+  std::vector<StrideB> stride_B_host;
+  std::vector<StrideC> stride_C_host;
+  std::vector<StrideD> stride_D_host;
+
+  std::vector<ElementAccumulator> alpha_host;
+  std::vector<ElementAccumulator> beta_host;
+
+  // Device-side allocations
+  cutlass::DeviceAllocation<typename ProblemShape::UnderlyingProblemShape> problem_sizes;
+
+  // This example defines all matrices in a single allocation (e.g. block_A), but this is not a
+  // requirement. Matrix base pointers are read from device allocation (e.g. ptr_A)
+  cutlass::DeviceAllocation<ElementA> block_A;
+  cutlass::DeviceAllocation<ElementB> block_B;
+  cutlass::DeviceAllocation<ElementC> block_C;
+  cutlass::DeviceAllocation<ElementOutput> block_D;
+  cutlass::DeviceAllocation<ElementOutput> block_ref_D;
+
+  cutlass::DeviceAllocation<const ElementA *> ptr_A;
+  cutlass::DeviceAllocation<const ElementB *> ptr_B;
+  cutlass::DeviceAllocation<const ElementC *> ptr_C;
+  cutlass::DeviceAllocation<ElementOutput *> ptr_D;
+  cutlass::DeviceAllocation<ElementOutput *> ptr_ref_D;
+
+  cutlass::DeviceAllocation<StrideA> stride_A;
+  cutlass::DeviceAllocation<StrideB> stride_B;
+  cutlass::DeviceAllocation<StrideC> stride_C;
+  cutlass::DeviceAllocation<StrideD> stride_D;
+
+  // Note, this is an array of pointers to alpha and beta scaling values per group
+  cutlass::DeviceAllocation<ElementAccumulator*> alpha_device;
+  cutlass::DeviceAllocation<ElementAccumulator*> beta_device;
+  cutlass::DeviceAllocation<ElementAccumulator> block_alpha;
+  cutlass::DeviceAllocation<ElementAccumulator> block_beta;
+
+  uint64_t seed = 0;
+
+  //
+  // Methods
+  //
+  template <typename SrcT, typename DstT>
+  void convert_fp8_to_fp16(const SrcT* d_src, DstT* d_dst, size_t size) {
+      SrcT* h_src = new SrcT[size];
+      syclcompat::memcpy(h_src, d_src, size * sizeof(SrcT));
+      syclcompat::wait();
+      
+      DstT* h_dst = new DstT[size];
+      for (size_t i = 0; i < size; ++i) {
+          h_dst[i] = static_cast<DstT>(h_src[i]);
+      }
+
+      syclcompat::memcpy(d_dst, h_dst, size * sizeof(DstT));
+      syclcompat::wait();
+  }
```

Comment:
```suggestion
 template <typename SrcT, typename DstT>
  void convert_fp8_to_fp16(const SrcT* d_src, DstT* d_dst, size_t size) {
    constexpr int fragment_size = std::is_same_v<SrcT, cute::float_e5m2_t> ? 4 : 8;
    syclcompat::get_default_queue().parallel_for(size, [=](auto indx) {
      d_dst[indx] = static_cast<DstT>(d_src[indx]);
    }).wait();
  }
```

Reply:
Thank you, @mehdi-goli! :)
I applied this review suggestion.

@min-jean-cho, please advise if you're okay with this change. I'm not aware of the reason why the implementation in the sycl-develop branch does `D2H copy -> FP8 to FP16 conversion -> H2D copy` for the reference implementation. It's not clear from the code. Thanks!